ext/opcache/jit/zend_jit_ir.c

```diff
@@ -7728,7 +7728,14 @@ static int zend_jit_bool_jmpznz(zend_jit_ctx *jit, const zend_op *opline, uint32
 			if_double = ir_IF(ir_EQ(type, ir_CONST_U8(IS_DOUBLE)));
 			ir_IF_TRUE(if_double);
 		}
-		ref = ir_NE(jit_Z_DVAL(jit, op1_addr), ir_CONST_DOUBLE(0.0));
+
```

Comment:
Need to avoid calling `zend_is_true()` during compilation above when the value is a const nan, otherwise we emit two warnings here: (during tracing, then while compiling the trace)
``` php
echo (bool)NAN;
```